students/AndrewMiotke/session07/HTML_Renderer/html_render.py

```diff
@@ -0,0 +1,129 @@
+#!/usr/bin/env python3
+
+"""
+A class-based system for rendering html.
+"""
+
+# This is the framework for the base class
+class Element: # Did we need the () in Python3+ ?
+
+    tag = 'html'
+    indent = "    "
+
+
+    def __init__(self, content=None, **kwargs):
+        self.attributes = kwargs
+        if content:
+            self.contents = [content]
+        else:
+            self.contents = []
+
+
+    def append(self, new_content):
+        self.contents.append(new_content)
+
+
+    def make_tags(self):
+        attributes = " ".join([f'{key}="{val}' for key, val in self.attributes.items()])
```

Comment:
This is throwing me an error when I try to run this, are you able to run this properly?

Reply:
What error are you getting? I'm not receiving an error but I'm also not getting any output 🤔

Reply:
Found the error, 
```
Traceback (most recent call last):
  File "run_html_render.py", line 46, in <module>
    render_page(page, "test_html_output1.html")
  File "run_html_render.py", line 27, in render_page
    page.render(f)
  File "/Users/andrewmiotke/GitHub/SP_2019_210A_classroom/students/AndrewMiotke/session07/HTML_Renderer/html_render.py", line 44, in render
    stuff.render(out_file, current_indent + self.indent)
AttributeError: 'str' object has no attribute 'render'
```

I can't seem to find the reason why. My best guess is that string doesn't know of render but I'm not sure why.

Reply:
It looks like the issue here is that you're running the render on your self.contents. The render is a function of the WebElement not a string

Reply:
It looks like the very first item in your "stuff" is a String. This needs to be an element.